Modules/_asynciomodule.c

```diff
@@ -3345,9 +3409,20 @@ _asyncio__register_task_impl(PyObject *module, PyObject *task)
 /*[clinic end generated code: output=8672dadd69a7d4e2 input=21075aaea14dfbad]*/
 {
     asyncio_state *state = get_asyncio_state(module);
-    if (register_task(state, task) < 0) {
+    if (Task_Check(state, task)) {
```

Comment:
sorry for going back and forth here, but now I wonder if there's ever a case that these register / unregister functions are ever called on native asyncio tasks?

the native implementation calls the static register / unregister functions directly, so these functions should only be called by 3rd party providers, no?

Reply:
Technically yes but third party implementations often do weird things and I want to make sure there is no change in behavior here so this function handles both native and non-native tasks and I think that it is fine to handle both regardless.